### PR TITLE
Add uganda example

### PR DESCRIPTION
**Pull request by wang-boyu**
An implementation of the [Uganda Example](https://github.com/abmgis/abmgis/tree/master/Chapter05-GIS/Models/UgandaExample) in Python, using [Mesa](https://github.com/projectmesa/mesa) and [Mesa-Geo](https://github.com/projectmesa/mesa-geo).

### CODE REVIEW

File: examples/uganda/uganda/space.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    pass
+
+
+class UgandaArea(GeoSpace):
+    def __init__(self, crs):
+        super().__init__(crs=crs)
+
+    def load_data(self, population_gzip_file, lake_zip_file, world_zip_file):
+        world_size = gpd.GeoDataFrame.from_file(world_zip_file)
+        with gzip.open(population_gzip_file, "rb") as population_file:
```
Comment (rht) [inline examples/uganda/uganda/space.py:40]: We might want to package those with statements into a function for easy reuse.

Comment (rht) [reply examples/uganda/uganda/space.py:40]: Not in this PR, but in the future.

**Comment by rht** [comment]
How many lines of code for the Python vs NetLogo? Just curious.

File: examples/uganda/uganda/model.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+class Person(GeoAgent):
+    MOBILITY_RANGE_X = 0.0
+    MOBILITY_RANGE_Y = 0.0
+
+    def __init__(self, unique_id, model, geometry, crs, img_coord):
+        super().__init__(unique_id, model, geometry, crs)
+        self.img_coord = img_coord
+
+    def set_random_world_coord(self):
```
Comment (rht) [inline examples/uganda/uganda/model.py:23]: It seems to be a common pattern to set random coord? Should we package it?

File: examples/uganda/uganda/model.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import math
+import random
+import uuid
+
+import numpy as np
+from mesa import Model
```
Comment (rht) [inline examples/uganda/uganda/model.py:6]: Need to be just `import mesa`, and the rest should be `mesa.Model`, etc.

**Comment by wang-boyu** [comment]
> How many lines of code for the Python vs NetLogo? Just curious.

It is 205 in Python vs. 68 in NetLogo : ) Our code is formatted with black whereas the NetLogo code isn't.

File: examples/uganda/uganda/model.py (hunk 3 above)
Comment (wang-boyu) [reply examples/uganda/uganda/model.py]: Updated the code accordingly.

File: examples/uganda/uganda/space.py (hunk 1 above)
Comment (wang-boyu) [reply examples/uganda/uganda/space.py:40]: Sure, have created an issue for it: https://github.com/projectmesa/mesa-geo/issues/91.

File: examples/uganda/uganda/model.py (hunk 2 above)
Comment (wang-boyu) [reply examples/uganda/uganda/model.py:22]: Also included in https://github.com/projectmesa/mesa-geo/issues/91.

**Pull request merged**